rs/http_endpoints/public/src/common.rs

```diff
@@ -246,12 +246,25 @@ impl IntoResponse for CborUserError {
     }
 }
 
-pub(crate) fn validation_error_to_http_error(
+pub(crate) fn validation_error_to_http_error<C: std::fmt::Debug>(
+    request: &HttpRequest<C>,
```

Comment:
nit: Optionally we could make this
```
pub(crate) fn validation_error_to_http_error<C: std::fmt::Debug + HttpRequestContent>(
    request: &HttpRequest<C>,
    err: RequestValidationError,
    log: &ReplicaLogger,
) -> HttpError {
```
i.e,. we don't need to take the `message_id` separately, because we can just get it from the request via `request.id()`.